### DISCUSSION TITLE
Resending raw IR codes with Tuya ZS06 problems

### DISCUSSION THREAD

**Discussion by fjuppe** [Koenkk/zigbee2mqtt#23784] [Devices]
Hello,
Just got my Tuya ZS06 

ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus
Zigbee2MQTT version
[1.39.1](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.39.1)
Type of Coordinator
 zStack3x0
Coordinatorrevision
 20231111
Coordinator IEEE-adress
 0x00124b002f8ac433
Frontend version
0.7.4
Zigbee-herdsman-converters version
19.72.0
Zigbee-herdsman version
 0.55.3

I learned it with an on/off code from my TV-remote:

`'{"ir_code_to_send":"BXsjvxE0AsABA6oGNALgCwFAF0ADQAFAB+AHA+ADAUAb4AcBQBPAA0ABwAvABwkRnnsj6Qg0Av//4AoHAgg0Ag=="}'`

and it works both within Z2M and with a mosquitto_pub script triggered from Domoticz. The message payload is different for every time the button on Remote control is pressed but the stored one above always works.

Then I used the same procedure (and the same rules apply with different codestrings every from my A/C IR remote but on/off button gives a much longer code:

`'{"ir_code_to_send":"B0UXHh3tAZAN4AEDA+8F7QHADwPvBaYBQAsB7wVAAwCmYAfgAQ/gAwsApmAXgBsBpgFAA+ABC8AXAe0BQANAF0ADAe8FgCPgAwsH7wXtAZANpgHgAweAC0AHAaYBgAcA7eAEB0ATQMMN//90I7IRYgLvBWIC7QHgFQNAI+APA+AHO0An4AMTQA/gAwNAG+ADDxIhnHQjqghiAv//kA3vBe0BdASmIAMJ7QEqAaYBEAHtAUADQA9ABwMqAe0BQAsBdASAG8AH4AsXA3QE7QFAM0ADA3QEpgFAB0ADA3QE7QFALwEqAeABEwMqAe0BQBtAA+APC0AbwAMCdASmIAMF7QEqAaYBQBNAC0AHQAMCdASmYANAC0ADAyoB7QFAB0ADQAsDdATtAUAHQA/gCwNAHwF0BIA/AXQE4AEDQCvgAzfgDwtAJ+ADAwEqAYBLQBMDKgHtAUAHQAPgAwtAD+APAwEqAeAJP0CXQC9AB+ADA0ATCf//kA3kBqYBdASAA+ADAUATwAFAC0ADQAHAB0ABQAtAA0ABQAdAA8AB4A8L4DMB4A9TQBdAA+APAeATG0ABQB9AA0ABQAfAA+CLAcCbC3QEpgF0BKYBpgGmAQ=="}'`

This code newer has worked ( I have tried many times with different codes.)
MQTT Explorer shows there are differences in the received string after about 200 characters.
Is there a limitation somewhere in the system that sets a limit ob the payload ??
Or is somebody aware of why this longer string can not be sent back correctly either through Z2M or moquitto/MQTT Explorer

Thanks in advance for your inputs!

**Comment by raphael1688** [comment]
If you use SmartIR with ZS06 via MQTT, you may need to modify controller.py for MQTT class to remove the escape character before sending IR codes to Z2M.

Hope this will help.

![image](https://github.com/user-attachments/assets/ab1f072d-edb9-41c2-bfc6-114b063f9b2c)

**Comment by BjornPython** [comment]
> This code newer has worked


Did you mean it has never worked?


Were you able to make this work?
